src/commands/createFunction/scriptSteps/ScriptFunctionCreateStep.ts

```diff
@@ -56,8 +62,34 @@ export class ScriptFunctionCreateStep extends FunctionCreateStepBase<IScriptFunc
 
         const language: ProjectLanguage = nonNullProp(context, 'language');
         const fileName: string | undefined = getScriptFileNameFromLanguage(language);
+        await this._installDurableDependenciesIfNeeded(context, language);
+
         return fileName ? path.join(functionPath, fileName) : functionJsonPath;
     }
 
+    private async _installDurableDependenciesIfNeeded(context: IScriptFunctionWizardContext, language: ProjectLanguage): Promise<void> {
+        if (!context.newDurableStorageType) {
+            return;
+        }
+
+        const dfDepInstallFailed: string = localize('failedToAddDurableDependency', 'Failed to add or install durable package dependency. Please inspect and verify if it needs to be added manually.');
+
+        try {
+            switch (language) {
+                case ProjectLanguage.JavaScript:
+                case ProjectLanguage.TypeScript:
+                    await cpUtils.executeCommand(ext.outputChannel, context.projectPath, 'npm', 'install', durableUtils.nodeDfPackage);
+                    break;
+                case ProjectLanguage.Python:
+                    await pythonUtils.addDependencyToRequirements(durableUtils.pythonDfPackage);
+                    await venvUtils.runPipInstallCommandIfPossible();
+                    break;
+                default:
+            }
+        } catch {
+            ext.outputChannel.appendLog(dfDepInstallFailed);
```

Comment:
Hmmm, do we want to show a more detailed error message in the logs?  I don't think I'd want to re-throw the error in this cause just because there's already instructions in the template file on how to install the dependency, so I feel there's no need to block the user if at all possible since this is more of a 'nice-to-have' than a 'need-to-have' enhancement.

Reply:
I said it in another comment, but I think it's still worth outputting the original error with the localized string as well.